### PR TITLE
replace string with enum for language 

### PR DESCRIPTION
**Pull request by ashtul**
language_t instead of char *

### CODE REVIEW

**Comment by ashtul** [comment]
fix #1073

File: src/stemmer.c [hunk 1]
```diff
@@ -6,19 +6,74 @@
...
-    if (!strncasecmp(language, __supportedLanguages[i],
-                     MAX(len, strlen(__supportedLanguages[i])))) {
-      return 1;
+typedef struct langPair_s
+{
+  const char *str;
+  language_t lang;
+} langPair_t;
+
+langPair_t __langPairs[] = {
```
Comment (mnunberg) [inline src/stemmer.c]: Please prefix the enums with `RS_`, e.g. `RS_LANG_EN` or `RS_LANG_FR`, etc.

File: src/stemmer.c [hunk 2]
```diff
@@ -63,15 +118,16 @@ void __sbstemmer_Free(Stemmer *s) {
...
+  if (type != stemmer->type || stemmer->language == UNSUPPORTED_LANGUAGE ||
+                               stemmer->language == language) {
     return 0;
   }
   return 1;
 }
 
-Stemmer *__newSnowballStemmer(const char *language) {
-  struct sb_stemmer *sb = sb_stemmer_new(language, NULL);
+Stemmer *__newSnowballStemmer(language_t language) {
```
Comment (mnunberg) [inline src/stemmer.c]: RSLanguage, not `language_t`. Also, `_t` is reserved for system types.

File: src/stemmer.c [hunk 3]
```diff
@@ -6,19 +6,74 @@
...
+  { "russian",    RUSSIAN },
+  { "spanish",    SPANISH },
+  { "swedish",    SWEDISH },
+  { "tamil",      TAMIL },
+  { "turkish",    TURKISH },
+  { "chinese",    CHINESE },
+  { NULL,         UNSUPPORTED_LANGUAGE }
+};
+
+const char *GetLanguageStr(language_t language) {
```
Comment (mnunberg) [inline src/stemmer.c]: RSLanguage_ToString

File: src/stemmer.c [hunk 4]
```diff
@@ -6,19 +6,74 @@
...
+    case TAMIL:       ret = "tamil";      break;
+    case TURKISH:     ret = "turkish";    break;
+    case CHINESE:     ret = "chinese";    break;
+    case UNSUPPORTED_LANGUAGE:  
+    default: break;
+  }
+  return (const char *)ret;
+}
+
+language_t GetLanguageEnum(const char *language) {
```
Comment (mnunberg) [inline src/stemmer.c]: RSLanguage_Find()

File: src/stemmer.c [hunk 5]
```diff
@@ -6,19 +6,74 @@
...
+  if (language == NULL)
+    return DEFAULT_LANGUAGE;
+
+  for (int i = 0; __langPairs[i].str != NULL; i++) {
```
Comment (mnunberg) [inline src/stemmer.c]: size_t

File: src/stemmer.c (hunk 1 above)
Comment (ashtul) [reply src/stemmer.c]: @mnunberg 
for English and French it is clear but do we want RS_LANG_DA or RS_LANG_DU?
I think it is better to keep RS_LANG_DANISH.

File: src/aggregate/aggregate_request.c [hunk 6]
```diff
@@ -286,6 +286,7 @@ static int parseQueryLegacyArgs(ArgsCursor *ac, RSSearchOptions *options, QueryE
 static int parseQueryArgs(ArgsCursor *ac, AREQ *req, RSSearchOptions *searchOpts,
                           AggregatePlan *plan, QueryError *status) {
   // Parse query-specific arguments..
+  char *languageStr = NULL;
```
Comment (mnunberg) [inline src/aggregate/aggregate_request.c]: const char

File: src/stemmer.c (hunk 1 above)
Comment (mnunberg) [reply src/stemmer.c]: true

Comment (mnunberg) [reply src/stemmer.c]: well, it'd be DK and NL ;)

File: src/stemmer.h [hunk 7]
```diff
@@ -6,9 +6,31 @@
 extern "C" {
 #endif
 
+typedef enum LANGUAGE {
```
Comment (mnunberg) [inline src/stemmer.h]: no need for enum label here

Comment (ashtul) [reply src/stemmer.h]: It fails without. You might mix C and C++

**Pull request merged**